### PR TITLE
heroic: 2.9.1 -> 2.9.2

### PR DESCRIPTION
**Pull request by aidalgol**
## Description of changes
* Update Heroic Games Launcher to `2.9.2`.
* Add new Heroic dependency `vulkan-helper`.

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/vulkan-helper/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, vulkan-loader
+, addOpenGLRunpath
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "vulkan-helper";
+  version = "2023-09-16";
```
Comment (K900) [inline pkgs/tools/graphics/vulkan-helper/default.nix:10]: `unstable-2023-09-16`

File: pkgs/tools/graphics/vulkan-helper/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  cargoSha256 = "sha256-s5QytuNhjZQhIDJtpeAW3J4op1t4nC+xD2i7Zf5mzfw=";
+
+  nativeBuildInputs = [
+    addOpenGLRunpath
+  ];
+
+  postFixup = ''
+    patchelf --add-rpath ${vulkan-loader}/lib $out/bin/vulkan-helper
+    addOpenGLRunpath $out/bin/vulkan-helper
```
Comment (K900) [inline pkgs/tools/graphics/vulkan-helper/default.nix:27]: Are you sure this is necessary? The loader alone _should_ be enough.

File: pkgs/tools/graphics/vulkan-helper/default.nix (hunk 1 above)
Comment (aidalgol) [reply pkgs/tools/graphics/vulkan-helper/default.nix:10]: Is "unstable" the right term, since upstream doesn't do releases at all?

Comment (K900) [reply pkgs/tools/graphics/vulkan-helper/default.nix:10]: It's the convention. /shrug

File: pkgs/tools/graphics/vulkan-helper/default.nix (hunk 2 above)
Comment (aidalgol) [reply pkgs/tools/graphics/vulkan-helper/default.nix:27]: One of the program's subcommands crashes trying to `dlopen` an OpenGL library.

Comment (K900) [reply pkgs/tools/graphics/vulkan-helper/default.nix:27]: Fun.

**Review by K900 — APPROVED** [review]
Tested on my Steam Deck, runs fine.

File: pkgs/tools/graphics/vulkan-helper/default.nix (hunk 1 above)
Comment (aidalgol) [reply pkgs/tools/graphics/vulkan-helper/default.nix:10]: Done.

**Pull request merged**